{
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# Analysing phylogenetic trees"
      ],
      "metadata": {
        "id": "dZgBTr0HVn3T"
      },
      "id": "dZgBTr0HVn3T"
    },
    {
      "cell_type": "markdown",
      "source": [
        "This module has been adapted from the Wellcome Connecting Science Antimicrobial Resistance (AMR) Africa 2024 Course.\n",
        "\n",
        "\n",
        "**Module Developers:** Dr. Stanford Kwenda and Dr Francesc Coll I Cerezo\n",
        "\n",
        "Source: https://github.com/WCSCourses/AMR-Africa-24"
      ],
      "metadata": {
        "id": "Y1tqHDhosWjE"
      },
      "id": "Y1tqHDhosWjE"
    },
    {
      "cell_type": "markdown",
      "source": [
        "# What are phylogenetic trees?\n",
        "\n",
        "A phylogeny, also known as phylogenetic tree, depicts estimated\n",
        "evolutionary relationships between taxa - these can be species, strains\n",
        "or even genes.\n",
        "\n",
        "In the context of infectious diseases epidemiology, phylogenetic trees\n",
        "are commonly used to define evolutionary relationships between strains\n",
        "of the same bacterial species. This is possible because bacteria\n",
        "reproduce clonally.\n",
        "\n",
        "During clonal reproduction, bacterial progenitor cells replicate their\n",
        "DNA at high fidelity. Despite this, random errors in DNA replication may\n",
        "still occur, resulting in a clonal progeny that will inherit these\n",
        "genetic replication 'errors' (i.e. mutations) in their DNA and may not\n",
        "be strictly identical to their progenitor cells. Bacterial strains that\n",
        "have recently originated from the same progenitor cell are thus expected\n",
        "to share identical genomes, or have diverged at most by only a few\n",
        "genetic differences (mutations). The number and pattern of shared\n",
        "mutations between bacterial strains can be used to reconstruct their\n",
        "genealogical and evolutionary relationships.\n",
        "\n",
        "On a phylogenetic tree, isolated bacterial strains are depicted on the\n",
        "tips (or leaves) of the tree (i.e. taxa), whereas the internal nodes of\n",
        "the tree denote their hypothetical ancestors. Nodes and taxa are\n",
        "connected by branches, the length of which represent genetic distances\n",
        "between connected groups. Groups of bacterial strains (taxa) that share\n",
        "the same common ancestor form a monophyletic group (also known as\n",
        "clade). A group of strains that descends from a common ancestor, but\n",
        "does not include all descendants, is called paraphyletic."
      ],
      "metadata": {
        "id": "3ZjvXRxxsSzC"
      },
      "id": "3ZjvXRxxsSzC"
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "# How are phylogenetic trees reconstructed?\n",
        "\n",
        "Today almost all phylogenetic trees are inferred from molecular sequence\n",
        "data, most often from DNA sequences. This is because DNA is an inherited\n",
        "material; it can easily, reliably and inexpensively be extracted and\n",
        "sequenced; and DNA sequences are highly specific to bacterial species\n",
        "and strains.\n",
        "\n",
        "The application of whole-genome sequencing now makes it possible to\n",
        "'read' the DNA sequence of the whole bacterial chromosome, which\n",
        "provides the ultimate level of resolution possible to discriminate\n",
        "between closely related strains. The figure below illustrates the common\n",
        "workflow to generate multiple DNA sequence alignments from a collection\n",
        "of bacterial strains. Generally, bacterial DNA is extracted from a\n",
        "single colony picked from culture plates (referred as to 'isolate'),\n",
        "followed by library preparation and whole-genome sequencing using rapid\n",
        "benchtop sequencers. Raw sequence data generated by sequencers are\n",
        "processed using bioinformatic and genomic pipelines, which generally\n",
        "involve mapping (aligning) the generated short reads to a reference\n",
        "genome to reconstruct the isolate's DNA sequence along the whole\n",
        "bacterial chromosome. Mapping the short reads of multiple sequenced\n",
        "isolates to the same reference genome allows the creation of multiple\n",
        "sequence alignments. Multiple sequence alignments are the first and\n",
        "critical point from which phylogenetic trees can be re-constructed.\n",
        "\n",
        "![image explained in more detail in the article\n",
        "text](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-000.png)\n",
        "\n",
        "Polymorphic sites (that is, nucleotide positions that are variable\n",
        "across multiple strains) in multiple alignments are used to infer\n",
        "evolutionary relationships, whereas monomorphic sites (nucleotide\n",
        "positions with the same DNA base) are generally ignored. The figure\n",
        "below shows an example of a simple multiple alignment of eight sites\n",
        "from four strains, which include monomorphic (squared) and polymorphic\n",
        "sites. Genetic changes in the phylogenetic tree are showed as coloured\n",
        "vertical rectangles on the branch where they originated. The\n",
        "identification of genetic changes (alleles) that are unique and common\n",
        "to multiple taxa (strains) are used to group them into monophyletic\n",
        "groups (clades) in a hierarchical manner (see example below) with the\n",
        "goal of constructing the most plausible genealogical relationships\n",
        "between strains and clades.\n",
        "\n",
        "![image explained in more details in the article\n",
        "text](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-002.png)\n"
      ],
      "metadata": {
        "id": "c1ICf8h66Zzj"
      },
      "id": "c1ICf8h66Zzj"
    },
    {
      "cell_type": "markdown",
      "source": [
        "# How are phylogenetic trees used in infectious diseases epidemiology?\n",
        "\n",
        "Phylogenetic trees are commonly used to identify where person-to-person\n",
        "transmission occurs; to identify the sources and study the transmission\n",
        "routes of outbreak and epidemic clones; and to determine whether\n",
        "bacterial clones are restricted to specific hosts and settings or, on\n",
        "the contrary, able to circulate among multiple ones.\n",
        "\n",
        "A common phylogenetic method used to study how bacterial characteristics\n",
        "(traits) evolved is ancestral state reconstruction. In the example shown\n",
        "below, strains on the same tree are labelled based on the presence of\n",
        "different traits. Arrows indicate what internal node (ancestor) in the\n",
        "tree most likely changed (lost or gained) such a trait. Bacterial traits\n",
        "we may be interested in reconstructing include: geographical location -\n",
        "to then identify movement between regions (transmission events;\n",
        "colonising or infecting host - to enable us to identify host jumps; and\n",
        "antibiotic susceptibility - to enable us to identify evolution of AMR.\n",
        "The emergence and spread of individual mutations, genes and mobile\n",
        "genetic elements can also be reconstructed in a bacterial phylogeny\n",
        "using this method (see figure below).\n",
        "\n",
        "![image explained in more details in the article\n",
        "text](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-004.png)"
      ],
      "metadata": {
        "id": "2gxCSd6K6Vx6"
      },
      "id": "2gxCSd6K6Vx6"
    },
    {
      "cell_type": "markdown",
      "source": [
        "# How are phylogenetic trees interpreted?\n",
        "\n",
        "The preferred interpretation of a phylogenetic tree is as a depiction of\n",
        "lines of descent. That is, trees communicate the evolutionary\n",
        "relationships among strains and clades. Under this interpretation,\n",
        "internal nodes on a tree are taken to correspond to bacterial strains\n",
        "that existed in the past (ancestral) but could not be sampled.\n",
        "\n",
        "Phylogenies are commonly mis-interpreted when read along the tips.\n",
        "Instead, the correct way to read a tree is as a set of hierarchically\n",
        "nested groups (clades).\n",
        "\n",
        "In the tree above, strain C is more closely related to strain B than it\n",
        "is to strain A. This is inferred by tracing the ancestor of strains\n",
        "(depicted as internal nodes) using the branch structure (i.e. topology)\n",
        "of the tree. Relatedness should be understood in terms of common\n",
        "ancestry--- the more recently strains share a common ancestor, the more\n",
        "closely related they are."
      ],
      "metadata": {
        "id": "pDAbgBRD6lBo"
      },
      "id": "pDAbgBRD6lBo"
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Exercises on interpreting phylogenetic trees\n",
        "\n",
        "This section includes questions on inferring genetic relatedness using\n",
        "phylogenetic trees.\n",
        "\n",
        "![Diagram Description automatically\n",
        "generated](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-006.png)\n",
        "\n",
        "Question 1. based on the tree above, what\n",
        "internal node corresponds to the most recent common ancestor of samples\n",
        "8 and 10:\n",
        "\n",
        "-   Node F\n",
        "\n",
        "-   Node D\n",
        "\n",
        "-   Sample 7\n",
        "\n",
        "-   Node E\n",
        "\n",
        "Question 2. Based on the tree above, which group of samples are most\n",
        "closely related:\n",
        "\n",
        "-   Samples 1 to 5\n",
        "\n",
        "-   Samples 6 & 7\n",
        "\n",
        "-   Samples 6 to 10\n",
        "\n",
        "-   Samples 8 & 9\n",
        "\n",
        "Question 3. Based on the tree above, which of the following statements\n",
        "referring to sample 10 is more accurate:\n",
        "\n",
        "-   Sample 10 is more closely related to sample 7 than to sample 8\n",
        "\n",
        "-   Sample 10 is more closely related to sample 8 than to sample 7\n",
        "\n",
        "-   Sample 10 is equally related to sample 7 and sample 8\n",
        "\n",
        "-   Sample 10 is related to sample 8, but it is not related to sample 7\n",
        "\n",
        "Question 4. Based on the tree above, which of the following statements\n",
        "referring to sample 7 is more accurate:\n",
        "\n",
        "-   Sample 7 is more closely related to sample 8 than to sample 10\n",
        "\n",
        "-   Sample 7 is more closely related to sample 10 than to sample 8\n",
        "\n",
        "-   Sample 7 is equally related to sample 8 and sample 10\n",
        "\n",
        "-   Sample 7 is related to sample 8, but it is not related to sample 10\n",
        "\n",
        "![Diagram Description automatically\n",
        "generated](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-007.png)\n",
        "\n",
        "Question 5. Based on the country of origin of samples on the tree above,\n",
        "which of the following statements about transmission events is more\n",
        "certain:\n",
        "\n",
        "-   The common ancestor of samples 6 to 10 (node D) most likely circulated in country A first and later on transmitted to country B and C\n",
        "\n",
        "-   The common ancestor of samples 6 to 10 (node D) most likely circulated in country B first and later on transmitted to country C\n",
        "\n",
        "-   The common ancestor of samples 6 to 10 (node D) most likely circulated in country C first and later on transmitted to country B\n",
        "\n",
        "-   The common ancestor of samples 6 to 10 (node D) could have circulated in country A or B\n",
        "\n",
        "Question 6. Based on the country of origin of samples on the tree above,\n",
        "which of the following statements about transmission events is more\n",
        "certain:\n",
        "\n",
        "-   The common ancestor of samples 1 to 10 (node A) most likely circulated in country A first and later on transmitted to country B and C\n",
        "\n",
        "-   The common ancestor of samples 1 to 10 (node A) most likely circulated in country B first and later on transmitted to country A and C\n",
        "\n",
        "-   The common ancestor of samples 1 to 10 (node A) most likely circulated in country C first and later on transmitted to country A and B\n",
        "\n",
        "-   The common ancestor of samples 1 to 10 (node A) could have circulated in country A or B\n",
        "\n"
      ],
      "metadata": {
        "id": "MzbuFn-09VCK"
      },
      "id": "MzbuFn-09VCK"
    },
    {
      "cell_type": "code",
      "source": [
        "# Enter the answers to the Qs above below\n",
        "# 1 -\n",
        "# 2 -\n",
        "# 3 -\n",
        "# 4 -\n",
        "# 5 -\n",
        "# 6 -"
      ],
      "metadata": {
        "id": "MHBgESnUI8AA"
      },
      "id": "MHBgESnUI8AA",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Exercises on constructing a phylogeny from gene sequences and whole genome SNPs\n",
        "\n",
        "Having learned how to interpret the tree, we will next construct\n",
        "phylogenetic trees using the sequence from gene *mecX*, an imaginary\n",
        "hypothetical gene from *Staphylococcus aureus*. *MecX* has a high\n",
        "sequence similarity to *mecA*, a gene identified as responsible for\n",
        "methicillin resistance in *Staphylococcaceae* family. In this practical,\n",
        "you will compare and contrast the phylogenetic relationship of the gene\n",
        "*mecX* and the whole genome of *Staphylococcus aureus*. Both the *mecX*\n",
        "sequences, and the whole genome SNP alignment have been provided for you\n",
        "in the session folder. The whole genome SNP alignment was subsampled\n",
        "from the data you generated in the previous session. To ensure that the\n",
        "phylogenetic tree can be constructed in a timely manner, we will be\n",
        "working with 13 *S. aureus* genomes and their *mecX* genes.\n",
        "\n",
        "The *mecX* is carried by the staphylococcal chromosome cassette *mec*\n",
        "(SCC*mec*), a mobile genetic element highlighted as recombination\n",
        "hotspot. The origins of *mecX* remained unclear with two competing\n",
        "hypotheses: (i) *mecX* was passed on vertically from generation to\n",
        "generation; and (ii) *mecX* was repeatedly horizontally acquired by\n",
        "independent recombination events. To address the hypothesis, you will\n",
        "construct phylogeny from *mecX* sequences and whole genome SNP of *S.\n",
        "aureus*. The topology of the gene tree versus the whole genome SNPs tree\n",
        "will give you ideas of how gene X dissemination in the population"
      ],
      "metadata": {
        "id": "3GB8zqbRFLeu"
      },
      "id": "3GB8zqbRFLeu"
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "# Viewing the alignment in SeaView\n",
        "\n",
        "You will use SeaView to view and edit alignments, as well as to make a\n",
        "phylogeny. The program has a graphical user interface (GUI) that is easy\n",
        "to operate.\n",
        "\n",
        "First - download seaview from this website:\n",
        "https://doua.prabi.fr/software/seaview\n",
        "\n",
        "and choose your operating system.\n",
        "\n",
        "## retrieve the data\n",
        "Retrieve the data for this practical by clicking or pasting this link into your browser:\n",
        "https://wcs_data_transfer.cog.sanger.ac.uk/phylogeny_prac_cp9.zip\n",
        "\n",
        "and uncompress the files\n",
        "\n",
        "## start Seaview\n",
        "\n",
        "You may start seaview by clicking the executable for your respective operating system.\n",
        "\n",
        "## load the alignment\n",
        "Load the alignment of *mecX* sequences by selecting \"Open\" from the \"File\" menu and selecting the \"mecX.sequence.fa\" file.\n",
        "\n",
        "![Graphical user interface Description automatically\n",
        "generated](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-008.png)\n",
        "\n",
        "Please spend some time to look at the diversity of *mecX* sequences in\n",
        "your samples. *MecX* is diverse with high SNP density and several\n",
        "indels, which makes mapping challenging and prone to errors. Therefore,\n",
        "we will use the assembly version of the gene in this part of the\n",
        "practical.\n",
        "\n",
        "## Multiple sequence alignment\n",
        "\n",
        "It is important to make sure that the column in our data represent\n",
        "homologous bases by aligning the nucleotide or amino acid using a\n",
        "multiple alignment programme. For *mecX* sequences, length differences\n",
        "might complicate multiple sequence alignment as these require insertion\n",
        "or gaps into an alignment to ensure that homologous sites remain\n",
        "aligned. When possible, please check an alignment by eyes.\n",
        "\n",
        "![Graphical user interface, application Description automatically\n",
        "generated](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-010.png)\n",
        "\n",
        "Seaview allows alignment using two programmes, clustal and muscle.\n",
        "Generally muscle is faster, and the protein alignments are of similar\n",
        "quality to clustal. In both cases, sequences are aligned by assigning\n",
        "costs to particular base changes and gaps insertions, with the optimal\n",
        "alignment having the lowest cost.\n",
        "\n",
        "To start the alignment, select \"Align\" then \"Align all\"\n",
        "\n",
        "![Graphical user interface, application Description automatically\n",
        "generated](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-012.png)\n",
        "\n",
        "When the alignment process is complete, SeaView will have inserted gaps\n",
        "into the sequences so that homologous sites are lined up in colunms.\n",
        "Please inspect the alignment. You should be able to see which sequences\n",
        "are most closely related. If an alignment is still misaligned when\n",
        "observed by eye, you can edit the alignment as necessary.\n",
        "\n",
        "![Graphical user interface Description automatically\n",
        "generated](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-014.png)"
      ],
      "metadata": {
        "id": "t6YtT5kb-ReK"
      },
      "id": "t6YtT5kb-ReK"
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Constructing a phylogeny from gene sequences using PhyML\n",
        "\n",
        "To estimate the phylogeny, we will use a programme called PhyML, which\n",
        "is already included in SeaView. PhyML includes a number of nucleotide\n",
        "substitution models ranging from the very simple (and could be\n",
        "unrealistic) to a complex one. PhyML uses maximum likelihood (ML) to\n",
        "estimate the tree. ML is more accurate than simpler approaches as it\n",
        "specifies an explicit evolutionary model to account for sources of\n",
        "homoplasy, while does not take too long to run. Moreover, it uses an\n",
        "optimality criterion (likelihood), which enable different trees to be\n",
        "compared. You can read more about the tree model from\n",
        "[https://www.nature.com/articles/nrg3186](https://www.nature.com/articles/nrg3186)\n",
        "\n",
        "Click \"Trees\" option and selection \"PhyML\"\n",
        "\n",
        "![Graphical user interface Description automatically\n",
        "generated](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-016.png)\n",
        "\n",
        "![Graphical user interface Description automatically\n",
        "generated](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-018.png)\n",
        "\n",
        "We will start by running a tree with choosing GTR model for substitution\n",
        "rates. (change view in props by unticking before). In the branch support, select \"None\" as we will look at this\n",
        "parameter later. If you choose a model to include different transition\n",
        "and transversion substitution rates, you can set the ratio at Ts/Tv\n",
        "ratio. In the \"Across site rate variation\" box, select \"None\", we will\n",
        "look at the rate ratio later. The last two boxes help specify how the\n",
        "programme will choose the tree to start from (useful for large\n",
        "data). We will leave them as defaults.\n",
        "\n",
        "Once the run has finished, click \"OK\". You will have a phylogenetic as\n",
        "in the figure.\n",
        "\n",
        "![Graphical user interface, application, table Description automatically\n",
        "generated](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-020.png)\n",
        "\n",
        "The type of analysis run and the loglikelihood are shown above the tree.\n",
        "\n",
        "You can alter the placement of the root by selecting re-root option and\n",
        "clicking the node on the phylogeny. You can tick Br lengths to add\n",
        "branch length values to the tree.\n",
        "\n",
        "Please record likelihood score, that is ln(L) of your first tree.\n",
        "\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "CmOO8XsDBPDp"
      },
      "id": "CmOO8XsDBPDp"
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Creating gene trees with different models\n",
        "Not all sites within the sequence evolve at the same rate. Some parts of a gene evolve faster than others. For example, an active site of an enzyme which is functionally important may be more conserved than others. When comparing several sequences to estimate a phylogeny, we should account for rate heterogeneity to avoid errors.\n",
        "\n",
        "\n",
        "Select “Optimised” from the “Invariable sites” box to allow the proportion of invariant sites to change but keep other parameters the same as the previous analysis. Press “Run” to generate a phylogeny as before. Please record the likelihood score of your second tree.\n",
        "\n",
        "\n",
        "Comparing models with the likelihood ratio test (LRT)\n",
        "ln(L) model 1 =\n",
        "ln(L) model 2 =\n",
        "\n",
        "You can compare the likelihood scores of the first (simpler model) and the second tree (more complex model) to work out which model fits better.\n",
        "Alternatively, you can statistically test which model is better using the following formula:\n",
        "LR = 2 x (neg ln(L1) – neg ln(L2))\n",
        "Where neg ln(L1) and neg ln(L2) present the negative log likelihood of the simpler and more complex model, respectively.\n",
        "The significant of LR can be estimated using chi-square significant tables.\n"
      ],
      "metadata": {
        "id": "C72-M4QGMxB-"
      },
      "id": "C72-M4QGMxB-"
    },
    {
      "cell_type": "code",
      "source": [
        "# Complete the scores for the above:\n",
        "# ln(L) model 1 =\n",
        "\n",
        "# ln(L) model 2 ="
      ],
      "metadata": {
        "id": "hS0sY9EgIoKA"
      },
      "id": "hS0sY9EgIoKA",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "\n",
        "\n",
        "# Phylogeny estimation with bootstrapping\n",
        "\n",
        "Bootstrapping is a statistical technique to assess the confidence level\n",
        "around each phylogenetic node. The bootstrapping values indicates how\n",
        "many times out of 100, the same branch was observed when repeating the\n",
        "phylogenetic reconstruction on the re-sampled set of your data. Robust\n",
        "relationship should be repeatable, and subsequently observed in a large\n",
        "proportion of randomised data. Therefore, if you get 100 out of 100\n",
        "times for a particular node, you can be certain that the observed branch\n",
        "is not due to chance, but likely to be real.\n",
        "\n",
        "You can create a bootstrapped phylogeny for *mecX* dataset by creating a\n",
        "new phylogeny as before. This time click on \"Bootstrap\" in the \"Branch\n",
        "support\" box, and enter 10 in the replicates box. Ideally, you would\n",
        "need to run at least 100 or 1000 replicates. But due to speed and time\n",
        "limit, we will only work on 10 replicates in this exercise.\n",
        "\n",
        "![Graphical user interface, application Description automatically\n",
        "generated](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-022.png)"
      ],
      "metadata": {
        "id": "JukLjK6FBUxr"
      },
      "id": "JukLjK6FBUxr"
    },
    {
      "cell_type": "markdown",
      "id": "ed5686b8",
      "metadata": {
        "id": "ed5686b8"
      },
      "source": [
        "\n",
        "\n",
        "# Constructing phylogeny from whole genome SNPs\n",
        "\n",
        "You can either construct a phylogeny using a whole genome alignment\n",
        "produced from the previous practical, or using only sites that contain\n",
        "variations. The former will be slow to run. We therefore will only\n",
        "extract the sites which only contain SNPs using a C script called\n",
        "\"snp_sites\"\n",
        "([https://github.com/sanger-pathogens/snp-sites](https://github.com/sanger-pathogens/snp-sites)).\n",
        "\n",
        "The SNP alignment of *S. aureus* has been prepared for you. Open the SNP\n",
        "alignment in Seaview and make a tree as before. Do not include the\n",
        "invariant sites parameter, as we just removed all invariant sites from\n",
        "the dataset.\n",
        "\n",
        "![](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-024.png)\n",
        "\n",
        "Finally, you can compare the phylogenetic trees of *mecX* and *S.\n",
        "aureus* whole genome SNPs which have been produced for you to answer the\n",
        "following questions.\n",
        "\n",
        "![Chart, radar chart Description automatically\n",
        "generated](https://raw.githubusercontent.com/WCSCourses/ACORN-ClinAMR/main/course_data/6_June_Day_6/img/images-026.png)\n",
        "\n",
        "# Questions:\n",
        "\n",
        "1. How does the tree of whole genome SNPs compare to the *mecX* tree you\n",
        "have made?\n",
        "\n",
        "2. Why the analyses of different part of the genome (whole genome vs **mecX**) lead to different phylogenies?\n",
        "\n",
        "3. What does this tell you about the acquisition and dissemination of **mecX**?\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Enter the answers to Q1,2,3 above below\n"
      ],
      "metadata": {
        "id": "14rvBBYJCOZa"
      },
      "id": "14rvBBYJCOZa",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#click below for the answers to interpreting trees section"
      ],
      "metadata": {
        "id": "2AxudVipBqJX"
      },
      "id": "2AxudVipBqJX",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# @title Answers to exercises on interpreting phylogenetic trees:\n",
        "\n",
        "Question 1: 'Node E' is the correct answer. 'Node F' is an ancestor of\n",
        "sample 8 but not of sample 10. 'Node D' is a common ancestor of samples\n",
        "8 and 10, but it is more ancient common ancestor than 'node E'. 'Sample\n",
        "7' is a living specimen and is not an ancestor.\n",
        "\n",
        "Question 2: Samples 1 to 5 is the correct answer. Remember that in a\n",
        "tree represented as a rectangular layout, the length of horizontal lines\n",
        "(branches) represent genetic distances whereas vertical lines are only\n",
        "used to connect horizontal lines. In the tree above, samples 1 to 5 have\n",
        "the shortest branches connecting them to their common ancestor (node B).\n",
        "\n",
        "Question 3: Sample 10 is more closely related to sample 8 than to sample\n",
        "7. The most recent common ancestor of samples 10 and 8 is at node E,\n",
        "whereas the most common ancestor of samples 10 and 7 is at node D, which\n",
        "is a deeper (more ancestral) internal node in the tree.\n",
        "\n",
        "Question 4: Sample 7 is equally related to sample 8 and sample 10. The\n",
        "most recent common ancestor of samples 7 and 8 is at node D, as is the\n",
        "most recent common ancestor of sample 7 and 10. All descendants of node\n",
        "E are equally related to sample 7.\n",
        "\n",
        "Question 5: The common ancestor of samples 6 to 10 (node D) most likely\n",
        "circulated in country B first and later on transmitted to country C.\n",
        "Country B is the most likely origin of the common ancestor represented\n",
        "by 'node D' because its direct descendants ('node C' and 'node E') both\n",
        "contain samples collected on this country. Later on, one clone,\n",
        "represented by 'node F', transmitted from country B to C.\n",
        "\n",
        "Question 6: The common ancestor of samples 1 to 10 (node A) could have\n",
        "circulated in countries A or B. Unfortunately, information on the\n",
        "country of origin of contextual samples descendants from more ancestral\n",
        "nodes to 'Node A' are needed to draw more accurate conclusions."
      ],
      "metadata": {
        "cellView": "form",
        "id": "S0t2JvsKBx3S"
      },
      "id": "S0t2JvsKBx3S",
      "execution_count": null,
      "outputs": []
    }
  ],
  "metadata": {
    "jupytext": {
      "cell_metadata_filter": "-all",
      "main_language": "python",
      "notebook_metadata_filter": "-all"
    },
    "colab": {
      "provenance": []
    },
    "language_info": {
      "name": "python"
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}